core/sys/orca/orca.odin

```diff
@@ -0,0 +1,2183 @@
+}
+
+// A type representing a string list.
+str8_list :: struct {
+	// A linked-list of `oc_str8_elt`.
+	list: list,
+	// The number of elements in `list`.
+	eltCount: u64,
+	// The total length of the string list, which is the sum of the lengths over all elements.
+	len: u64,
+}
+
+// A type describing a string of 16-bits characters (typically used for UTF-16).
+str16 :: distinct []u16
+
+// A type representing an element of an `oc_str16` list.
+str16_elt :: struct {
+	// The string element is linked into its parent string list through this field.
+	listElt: list_elt,
+	// The string for this element.
+	_string: str16,
+}
+
+str16_list :: struct {
+	// A linked-list of `oc_str16_elt`.
+	list: list,
+	// The number of elements in `list`.
+	eltCount: u64,
+	// The total length of the string list, which is the sum of the lengths over all elements.
+	len: u64,
+}
+
+// A type describing a string of 32-bits characters (typically used for UTF-32 codepoints).
+str32 :: distinct []rune
+
+// A type representing an element of an `oc_str32` list.
+str32_elt :: struct {
+	// The string element is linked into its parent string list through this field.
+	listElt: list_elt,
+	// The string for this element.
+	_string: str32,
+}
+
+str32_list :: struct {
+	// A linked-list of `oc_str32_elt`.
+	list: list,
+	// The number of elements in `list`.
+	eltCount: u64,
+	// The total length of the string list, which is the sum of the lengths over all elements.
+	len: u64,
+}
+
+@(default_calling_convention="c", link_prefix="oc_")
+foreign {
+	// Make a string from a bytes buffer and a length.
+	str8_from_buffer :: proc(len: u64, buffer: [^]char) -> str8 ---
+	// Make a string from a slice of another string. The resulting string designates some subsequence of the input string.
+	str8_slice :: proc(s: str8, start: u64, end: u64) -> str8 ---
+	// Pushes a copy of a buffer to an arena, and makes a string refering to that copy.
+	str8_push_buffer :: proc(arena: ^arena, len: u64, buffer: [^]char) -> str8 ---
+	// Pushes a copy of a C null-terminated string to an arena, and makes a string referring to that copy.
+	str8_push_cstring :: proc(arena: ^arena, str: cstring) -> str8 ---
+	// Copy the contents of a string on an arena and make a new string referring to the copied bytes.
+	str8_push_copy :: proc(arena: ^arena, s: str8) -> str8 ---
+	// Make a copy of a string slice. This function copies a subsequence of the input string onto an arena, and returns a new string referring to the copied content.
+	str8_push_slice :: proc(arena: ^arena, s: str8, start: u64, end: u64) -> str8 ---
+	// Build a string from a null-terminated format string and a variadic argument list, similar to `vasprintf()`.
+	str8_pushfv :: proc(arena: ^arena, format: cstring, #c_vararg args: ..any) -> str8 ---
+	// Build a string from a null-terminated format string and variadic arguments, similar to `asprintf()`.
+	str8_pushf :: proc(arena: ^arena, format: cstring, #c_vararg args: ..any) -> str8 ---
```

Comment:
Probably a good idea to not allow these as primitives since it's C-style printf specifiers